docs/hpo/intro.md

```diff
@@ -0,0 +1,19 @@
+
+# Introduction
+
```

Comment:
This is not a great intro.
Maggy is:
* a programming model to simplify distributed machine learning
* a parallel execution engine for distributed machine learning
* an experiment tracking framework, if you run Maggy on Hopsworks

Maggy currently supports two execution engines:
* PySpark for distributed ML across many workers 
* Python for distributed ML, currently on a single worker


### Hello World Python

A hello-world Maggy program in Python is as follows:
def train():
  (...)
  return {"accuracy": 1.0 }
  
from maggy import experiment
experiment.lagom(train)

The above code will run the train function and register an experiment on Hopsworks with the accuracy=1.0 result.

### Hello World PySpark

def pytrain(config):
  iris = datasets.load_iris()
  X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, test_size=0.2, random_state=42)
  clf = XGBClassifier(max_depth=config.get("max_depth"))
  y_pred = clf.predict(X_test)
  accuracy = metrics.accuracy_score(y_test, y_pred)
  return {"accuracy": accuracy }
  
from maggy import experiment
config = Config ("max_depth" = [3. 4, 5])
experiment.lagom(pytrain, config)


### Maggy for Parallel Hyperparameter Tuning

Reply:
Are you describing the intro of Maggy or HPO?